be/src/olap/schema_change.cpp

```diff
@@ -1636,7 +1636,7 @@ OLAPStatus SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletRe
             break;
         }
         // set state to ready
-        WriteLock new_wlock(new_tablet->get_header_lock());
+        std::lock_guard<std::shared_mutex> new_wlock(new_tablet->get_header_lock());
```

Comment:
sometimes lock guard, sometimes unique_lock  what's the difference?

Reply:
std::lock_guard will be locked only once on construction and unlocked on destruction. but unique_lock has lock and unlock method, it is more flexible, condition_variable needs a unique_lock